src/langchain_google_spanner/vector_store.py

```diff
@@ -967,6 +1261,13 @@ def similarity_search_with_score(
         )
         return documents
 
+    @property
+    def __using_ANN(self):
```

Comment:
You also need to use this function in `max_marginal_relevance_search_with_score_by_vector` and `similarity_search_with_score_by_vector` and not go directly to KNN.

Reply:
@gauravpurohit06 I plan on doing that in a follow up PR. This one is already massive as it is and has taken 4+ weeks.